src/IK/src/InverseKinematics.cpp

```diff
@@ -529,3 +558,128 @@ bool HumanIK::initializeFloorContactTask(
+    ok = ok & m_jointRegularizationTask->initialize(taskHandler);
+    Eigen::VectorXd weightVector(m_kinDyn->getNrOfDegreesOfFreedom());
+    weightVector.setConstant(weight);
+    ok = ok & m_qpIK.addTask(m_jointRegularizationTask, taskName, 1, weightVector);
+
+    return ok;
+}
+
+bool HumanIK::initializeJointConstraintsTask(
+    const std::string& taskName,
+    const std::shared_ptr<BipedalLocomotion::ParametersHandler::IParametersHandler> taskHandler)
+{
+    bool ok{true};
+    bool useModelLimits;
+    if (!taskHandler->getParameter("use_model_limits", useModelLimits))
+    {
+        BiomechanicalAnalysis::log()->error("[HumanIK::initializeJointConstraintsTask] Parameter "
+                                            "'use_model_limits' of the {} task is missing",
+                                            taskName);
+        return false;
+    }
+    m_jointConstraintsTask = std::make_shared<BipedalLocomotion::IK::JointLimitsTask>();
+    ok = ok & m_jointConstraintsTask->setKinDyn(m_kinDyn);
+    double k_lim;
+    if (!taskHandler->getParameter("k_limits", k_lim))
+    {
+        BiomechanicalAnalysis::log()->error("[HumanIK::initializeJointConstraintsTask] Parameter "
+                                            "'k_limits' of the {} task is missing",
+                                            taskName);
+        return false;
+    }
+    std::vector<double> klim(m_kinDyn->getNrOfDegreesOfFreedom(), k_lim);
+    taskHandler->setParameter("klim", klim);
+    if (useModelLimits)
+    {
+        ok = ok & m_jointConstraintsTask->initialize(taskHandler);
+    } else
+    {
+        std::vector<std::string> jointNamesList;
+        if (!taskHandler->getParameter("joints_list", jointNamesList))
+        {
+            BiomechanicalAnalysis::log()->error("[HumanIK::initializeJointConstraintsTask] "
+                                                "Parameter "
+                                                "'joints_list' of the {} task is missing",
+                                                taskName);
+            return false;
+        }
+        std::vector<double> lowerBounds;
+        std::vector<double> upperBounds;
+        if (!taskHandler->getParameter("lower_bounds", lowerBounds)
+            || !taskHandler->getParameter("upper_bounds", upperBounds))
+        {
+            BiomechanicalAnalysis::log()->error("[HumanIK::initializeJointConstraintsTask] "
+                                                "Parameter "
+                                                "'lower_bounds' and/or 'upper_bounds of the {} "
+                                                "task is missing",
+                                                taskName);
+            return false;
+        }
+        if (jointNamesList.size() != lowerBounds.size()
+            || jointNamesList.size() != upperBounds.size())
+        {
+            BiomechanicalAnalysis::log()->error("[HumanIK::initializeJointConstraintsTask] "
```

Comment:
Because I saw from the HDE IK that the joint limits are set by default to the ones of the urdf model (see [here](https://github.com/robotology/human-dynamics-estimation/blob/cdf9c94238200ed30a348cbabe45b528ec0f7e72/HumanDynamicsEstimationLibrary/algorithms/src/DynamicalInverseKinematics.cpp#L1049-L1063)), and then the user can specify to set the joints limits only for some specified joints (see [here](https://github.com/robotology/human-dynamics-estimation/blob/cdf9c94238200ed30a348cbabe45b528ec0f7e72/HumanDynamicsEstimationLibrary/algorithms/src/DynamicalInverseKinematics.cpp#L1113-L1129))